### PR TITLE
HPCC-18200 Sort files by compressed/uncompressed in DFUQuery

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

**Review by afishbeck — APPROVED** [review]
@wangkx looks fine

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Comment by richardkchapman** [comment]
@jakesmith Any performance concerns?

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -12252,6 +12252,19 @@ IPropertyTreeIterator *deserializeFileAttrIterator(MemoryBuffer& mb, unsigned nu
...
+        void setIsCompressed(IPropertyTree* file)
+        {
+            if (isCompressed(*file))
+            {
+                file->setPropBool(getDFUQResultFieldName(DFUQRFiscompressed), true);
+                return;
+            }
+
+            const char * kind = file->queryProp(getDFUQResultFieldName(DFUQRFkind));
+            if (!isEmptyString(kind) && strieq(kind, "key"))
```
Comment (jakesmith) [inline dali/base/dadfs.cpp]: Using if (isFileKey(file)) would be better here.

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -12189,7 +12189,7 @@ IDFProtectedIterator *CDistributedFileDirectory::lookupProtectedFiles(const char
 
 const char* DFUQResultFieldNames[] = { "@name", "@description", "@group", "@kind", "@modified", "@job", "@owner",
     "@DFUSFrecordCount", "@recordCount", "@recordSize", "@DFUSFsize", "@size", "@workunit", "@DFUSFcluster", "@numsubfiles",
-    "@accessed", "@numparts", "@compressedSize", "@directory", "@partmask", "@superowners", "@persistent", "@protect" };
+    "@accessed", "@numparts", "@compressedSize", "@directory", "@partmask", "@superowners", "@persistent", "@protect", "@isCompressedFile" };
```
Comment (jakesmith) [inline dali/base/dadfs.cpp]: minor: "compressed" would be more concise/in keeping with other attribute names here.

**Review by jakesmith — COMMENTED** [review]
@wangkx - a couple of minor comments.

I don't have any efficiency concerns.

**Comment by wangkx** [comment]
@jakesmith please review my changes.

**Comment by jakesmith** [comment]
@wangkx - looks good.

@richardkchapman - ready to merge.

**Pull request merged**